src/main/java/main/Ui.java

```diff
@@ -0,0 +1,63 @@
+package main;
+
+import java.util.Scanner;
+
+public class Ui {
+    private Scanner scanner;
+
+    /**
+     * The default constructor for the Ui class
+     */
+    public Ui() {
+        this.scanner = new Scanner(System.in);
```

Comment:
```suggestion
        scanner = new Scanner(System.in);
```
Is "this" required here? I noticed the same issue in other parts of the code.